ffi/framebuffer_android.lua

```diff
@@ -3,14 +3,34 @@ local android = require("android")
 local BB = require("ffi/blitbuffer")
 local C = ffi.C
 
+--[[ configuration for devices with an electric paper display controller ]]--
+
+-- does the device has an e-ink screen?
 local has_eink_screen = android.isEink()
-local full, partial, a2, auto = 1, 2, 3, 4 -- luacheck: ignore
 
-local function update_eink(mode)
-    if has_eink_screen then
-	android.LOGV("requesting eink update " .. mode)
-        android.einkUpdate(mode)
+-- does the device needs to handle all screen refreshes
+local has_eink_full_support = android.isEinkFull()
+
+-- for *some* rockchip devices
+local rk_full, rk_partial, rk_a2, rk_auto = 1, 2, 3, 4 -- luacheck: ignore
+
+-- for *some* freescale devices
+local ntx_du, ntx_gc16, ntx_regal, ntx_full = 1, 2, 7, 34 -- luacheck: ignore
```

Comment:
On the off-chance that the first three match the [NTX headers I'm used to](https://github.com/NiLuJe/FBInk/blob/master/eink/mxcfb-kobo.h#L114), REAGL would be 6 (7 is REAGLD, which is iffier, requires an extra flag, and we generally don't need anyway).

Take that with a grain of salt, since that assumption breaks down with that 34 ;).

Reply:
> REAGL would be 6 (7 is REAGLD, which is iffier, requires an extra flag, and we generally don't need anyway).

True. Is `ntx_reagld` a good name?

> Take that with a grain of salt, since that assumption breaks down with that 34

Yeah, I based this PR on @char11 patch -> https://pastebin.com/rXgYYpaS and copied names from the comments there. As I don't see 34 I think it is just 32 (update_full) paired with 2 (mode_gc16) while ui refreshing is using 2 (0 from update_partial and 2 from mode_gc16). 

**warning** this post may contain broken logic

Reply:
Oh, right, then, no, according to those constants, GLR16 is REAGL, and is 7 indeed ;p.

I'd use AUTO (4) for UI, though, provided it behaves like I'm used to ;).

Reply:
And, yeah, according to the comments, that 34 now makes sense as full | gc16, yeah ;).

Reply:
> Oh, right, then, no, according to those constants, GLR16 is REAGL, and is 7 indeed ;p.
> 
> I'd use AUTO (4) for UI, though, provided it behaves like I'm used to ;).

Yeah, me too. And add 32 to ui flashing and partial flashing. But that would need a devce to test and current settings seem to do the work

Reply:
xref w/ https://github.com/koreader/koreader/issues/5056#issuecomment-500451752 if someone wants to test that theory ;).

Reply:
I tested the auto (4) mode - did not work.  Flashing should do a full refresh? I did not analyze it too much...

Reply:
Auto should not flash, no.

Reply:
So, TL;DR: What did you mean, exactly, by "did not work"?

Reply:
The result is mainly white screen, some artifacts.

Reply:
I changed FlashPartial and FlashUI to mode 39. It works.

Reply:
That AUTO behavior vaguely sounds like it always chooses A2 or DU.... Shame, that :/.

Reply:
I checked again. There is a logcat message imx6.gralloc: waveform_mode wrong with mode 4.

Reply:
Huh. I assume you pulled the constants from a decompiled JAR somewhere? Could there be some minor obfuscation that meant you got some of them wrong?

Reply:
These constants were in two different apk files.

Reply:
Curiouser and curiouser.

I assume that's on a device with explicit REGAL support? But that'd be a fairly harsh way of ensuring usage of REAGL instead of AUTO ;).

Reply:
Everything indicates that it has a regal support.